tests/interactive/test_get_all_possible_rewrites.py

```diff
@@ -0,0 +1,57 @@
+from xdsl.builder import ImplicitBuilder
+from xdsl.dialects import arith, func
+from xdsl.dialects.builtin import (
+    IndexType,
+    IntegerAttr,
+    ModuleOp,
+)
+from xdsl.interactive.get_all_possible_rewrites import (
+    IndexedIndividualRewrite,
+    IndividualRewrite,
+    get_all_possible_rewrites,
+)
+from xdsl.ir import Block, Region
+from xdsl.transforms import individual_rewrite
+
+
+def test_get_all_possible_rewrite():
+    # build module
+    index = IndexType()
+    module = ModuleOp(Region([Block()]))
+    with ImplicitBuilder(module.body):
+        function = func.FuncOp("hello", ((index,), (index,)))
+        with ImplicitBuilder(function.body) as (n,):
+            two = arith.Constant(IntegerAttr(0, index)).result
+            res = arith.Addi(two, n)
+            func.Return(res)
+
+    expected_res = (
+        (
+            IndexedIndividualRewrite(
+                3, IndividualRewrite(operation="arith.addi", pattern="AddImmediateZero")
+            )
+        ),
+    )
+
+    res = get_all_possible_rewrites(module, individual_rewrite.REWRITE_BY_NAMES)
+    assert res == expected_res
```

Comment:
Could you please add a custom rewrite defined in this file? So the module could be like this:
```
module {
    "test.op"() {"label" = "a"}
    "test.op"() {"label" = "a"}
    "test.op"() {"label" = "b"}
}
```
And the rewrites like this:
``` python
@dataclass
class Rewrite(RewritePattern):

    label: str

    @op_type_rewrite_pattern
    def match_and_rewrite(self, op: TestOp, rewriter: PatternRewriter):
        if op.attributes["label"] != StringAttr(self.label):
            return
        rewriter.replace(TestOp(attributes={"label": StringAttr("c")}))
```
The goal of this would be to have a test that is self-contained, and doesn't depend on dialects and rewrites that may change